### PR TITLE
Feature netmaker cli

### PR DESCRIPTION
**Pull request by alphadose**

### CODE REVIEW

File: cli/functions/http_client.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Fatalf("Client error making http request: %s", err)
+	}
+	resBodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		log.Fatalf("Client could not read response body: %s", err)
+	}
+	if res.StatusCode != http.StatusOK {
+		log.Fatalf("Error response: %s", string(resBodyBytes))
```
Comment (mattkasun) [inline cli/functions/http_client.go:74]: should probably return the http.Status or http.StatusCode here as well

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: cli/functions/server.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"net/http"
+
+	"github.com/gravitl/netmaker/cli/config"
+	cfg "github.com/gravitl/netmaker/config"
+	"github.com/gravitl/netmaker/models"
+)
+
+// GetLogs - fetch Netmaker server logs
+func GetLogs() string {
+	ctx := config.GetCurrentContext()
```
[the remark this answers is not in the document]
Comment (alphadose) [reply cli/functions/server.go]: done

**Review by mattkasun — APPROVED** [review]

**Review by 0xdcarns — APPROVED** [review]

File: cli/cmd/acl/deny.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		for _, node := range *functions.GetNodes(args[0]) {
+			nameIDMap[strings.ToLower(node.Name)] = node.ID
+		}
+		fromNodeID, ok := nameIDMap[strings.ToLower(args[1])]
+		if !ok {
+			log.Fatalf("Node %s doesn't exists", args[1])
+		}
+		toNodeID, ok := nameIDMap[strings.ToLower(args[2])]
+		if !ok {
+			log.Fatalf("Node %s doesn't exists", args[2])
```
Comment (abhishek9686) [inline cli/cmd/acl/deny.go:29]: I guess grammatically it should be -> Node %s doesn't **exist**
since these messages are displayed to the user

Comment (alphadose) [reply cli/cmd/acl/deny.go]: fixed

File: cli/cmd/user/create.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	},
+}
+
+func init() {
+	userCreateCmd.Flags().StringVar(&username, "name", "", "Name of the user")
+	userCreateCmd.Flags().StringVar(&password, "password", "", "Password of the user")
+	userCreateCmd.MarkFlagRequired("name")
+	userCreateCmd.MarkFlagRequired("password")
+	userCreateCmd.Flags().BoolVar(&admin, "admin", false, "Make the user an admin ?")
+	userCreateCmd.Flags().StringVar(&networks, "networks", "", "List of networks the user will access to (comma separated)")
```
Comment (abhishek9686) [inline cli/cmd/user/create.go:34]: `will have access to`

File: cli/cmd/user/update.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		} else {
+			user.Groups = []string{"*"}
+		}
+		functions.PrettyPrint(functions.UpdateUser(user))
+	},
+}
+
+func init() {
+	userUpdateCmd.Flags().BoolVar(&admin, "admin", false, "Make the user an admin ?")
+	userUpdateCmd.Flags().StringVar(&networks, "networks", "", "List of networks the user will access to (comma separated)")
```
Comment (abhishek9686) [inline cli/cmd/user/update.go:32]: `will have access to`

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**